### ISSUE TITLE
Freeze on completion

### ISSUE THREAD

**Issue by pmaziere** [artur-shaik/vim-javacomplete2#60]
Tested with 1ad698c , with and without YouCompleteMe enabled

I try to edit app/src/main/java/at/bitfire/icsdroid/ui/AddCalendarValidationFragment.java from https://gitlab.com/bitfireAT/icsdroid.git ( dd7777d ) and the only way to avoid the freeze when completing is to comment the "extends DialogFragment" part.

From what I can observe the vim cursor disappears and CPU usage goes 100% until I decide to kill javavi. Once javavi killed everything goes back to normal until the next completion attempt.

**Comment by artur-shaik** [comment]
It would be really nice, if you can prepare minimal project, that can reproduce your problem.
It's little bit hard for me to reproduce because I don't use gradle and I don't have those android libraries.

**Comment by DonnieWest** [comment]
Things are working fine on my end with that project, but there's some configuration involved.

@pmaziere have you tried loading your android.jar and support-v4.jar into the g:JavaComplete_LibsPath variable? If you're still getting issues afterward, please list the output of

```set g:JavaComplete_LibsPath?```

**Comment by pmaziere** [comment]
@artur-shaik : sorry, but this project is my first attempt to play with android and java in order to try to understand why it does not behave on my device. Therefore, I don't even know how to bootstrap such a project.

But I forgot to mention that I also installed hsanson/vim-android plugin.
So I disabled it and loaded the libs suggested by @DonnieWest in g:JavaComplete_LibsPath.
This resulted in a flawless completion.

So I checked how  hsanson/vim-android populates the CLASSPATH variable, and noticed that I was messing with this variable in my .vimrc.
Removing my mess fixed the issue.

Sorry for the noise
